packages/edit-site/src/components/post-list/index.js

```diff
@@ -353,6 +353,36 @@ export default function PostList( { postType } ) {
 		[ postTypeActions, editAction ]
 	);
 
+	const additionalContext = useSelect( ( select ) => {
+		const site = select( coreStore ).getEntityRecord( 'root', 'site' );
+		const themeTemplates = select( coreStore ).getEntityRecords(
+			'postType',
+			'wp_template',
+			{
+				per_page: -1,
+			}
+		);
+		const hasFrontPageTemplate = !! themeTemplates?.find(
+			( template ) => 'slug' in template && template.slug === 'front-page'
+		);
+		return {
+			siteSettings: {
```

Comment:
Please do not return new sub-objects in selectors, this will cause components to excessively re-render.

Reply:
I wonder if we could add a linting rule for this 🤔

Reply:
There is a runtime warning for this. The `useSelect` callback is called twice (with the same input data) and if both results are not shallow-equal, there is a console warning. This `siteSettings` object will surely trigger it.

Reply:
A lint rule would be even better, no?

Reply:
> Please do not return new sub-objects in selectors, this will cause components to excessively re-render.

Ah I see, I've removed the sub-objects here: https://github.com/WordPress/gutenberg/pull/65426/commits/6d36c1b8ee1c3c68601a61959bc1c2602ee107d1. Thanks 🙇 

A lint rule would be handy!